Library/Homebrew/cmd/--env.rb

```diff
@@ -1,20 +1,128 @@
 #:  * `--env`:
 #:    Show a summary of the Homebrew build environment.
 
+
 require "extend/ENV"
 require "build_environment"
 
+
 module Homebrew
+  
+  MAX_ITERATIONS = 10
+  VALID_SHELLS = %w(
+    sh csh tcsh bash ksh mksh oksh
+    pdksh zsh fish
+  )
+  DEFAULT_SHELL = "sh"
+
+
+  def canonicalize_shell(path)
+    # basename and strip leading '-' in login shell
+    name = File.basename path
+
+    if VALID_SHELLS.include? name
+      return name
+    end
+    if name.start_with? "-"
+      stripped_name = name.sub(/^[-]/, "")
+      if VALID_SHELLS.include? stripped_name
+        return stripped_name
+      end
+    end
+    return nil
```

Comment:
Don't need `return` as the last statement of a function in Ruby and `nil` is implicit if you haven't already passed something.